File: Day12/Car.py
class Car:
    def __init__(self):
        self.fuel_level = 0
        self.isRunning = False

    def get_fuel_level(self):
        return self.fuel_level
    def check_is_running(self):
        return self.isRunning

    def start_car(self):
        if self.fuel_level <= 0:
            raise ValueError("Car need to be fueled")
        self.isRunning = True

    def stop_car(self):
        self.isRunning = False

    def get_car_refuel(self, liters):
        if liters + self.fuel_level > 50:
            self.fuel_level = 50
        self.fuel_level += liters

    def drive_car(self,distance):
        if self.isRunning and self.fuel_level > 0:
            fuel_efficiency = 12
            fuel_consumed = distance /fuel_efficiency
            self.fuel_level = self.fuel_level - fuel_consumed

File: Day12/Car.js
class Car{
    constructor(){
        this.fuelLevel = 0;
        this.isRunning = false;
    }


    getFuelLevel(){
        return this.fuelLevel;
    }

    checkIsRunning(){
        return this.isRunning;

    }

    startCar(){
        if(this.fuelLevel < 0){
            throw new Error("Car need to be fueled");
        }
        this.isRunning = true;

    }

    stopCar(){
        this.isRunning = false;
    }

    refuelCar(liters){
        if((liters + this.fuelLevel) > 50){
            this.fuelLevel = 50;
        }
        this.fuelLevel += liters;

    }
    driveCar(distance){
        if(this.isRunning && this.fuelLevel > 0){
            let fuelEfficiency = 12;
            let fuelConsumed = distance / fuelEfficiency;
            this.fuelLevel -= fuelConsumed;
        }

    }




}
 
module.exports = Car